File: Génération_Code_Application/Cursor_AuraWear/frontend/src/types/systemEvent.ts
export type EventCategory = "auth" | "profile" | "transaction" | "admin";
export type EventSeverity = "info" | "warning" | "error";

export interface SystemEvent {
  id: number;
  event_type: string;
  category: EventCategory;
  severity: EventSeverity;
  success: boolean;
  user_id: number | null;
  actor_username: string | null;
  ip_address: string | null;
  user_agent: string | null;
  message: string;
  details: Record<string, unknown> | null;
  created_at: string;
}

export interface SystemEventListResponse {
  items: SystemEvent[];
  total: number;
  limit: number;
  offset: number;
}

export interface EventLogFilters {
  category?: EventCategory;
  event_type?: string;
  success?: boolean;
  user_id?: number;
  limit?: number;
  offset?: number;
}

File: Génération_Code_Application/Cursor_AuraWear/backend/app/models/system_event.py
from __future__ import annotations

import enum
from datetime import datetime
from typing import Any, Optional

from sqlalchemy import Boolean, DateTime, Integer, String, Text, func
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class EventCategory(str, enum.Enum):
    auth = "auth"
    profile = "profile"
    transaction = "transaction"
    admin = "admin"


class EventSeverity(str, enum.Enum):
    info = "info"
    warning = "warning"
    error = "error"


class SystemEvent(Base):
    __tablename__ = "system_events"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    event_type: Mapped[str] = mapped_column(String(80), index=True)
    category: Mapped[EventCategory] = mapped_column(index=True)
    severity: Mapped[EventSeverity] = mapped_column(default=EventSeverity.info)
    success: Mapped[bool] = mapped_column(Boolean, default=True)
    user_id: Mapped[Optional[int]] = mapped_column(Integer, nullable=True, index=True)
    actor_username: Mapped[Optional[str]] = mapped_column(String(50), nullable=True, index=True)
    ip_address: Mapped[Optional[str]] = mapped_column(String(45), nullable=True)
    user_agent: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    message: Mapped[str] = mapped_column(Text)
    details: Mapped[Optional[dict[str, Any]]] = mapped_column(JSONB, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        index=True,
    )
